src/main/scala/wasm/Parser.scala

```diff
@@ -420,6 +436,25 @@ class GSWasmVisitor extends WatParserBaseVisitor[WIR] {
     }
   }
 
+  override def visitHandlerInstr(ctx: HandlerInstrContext): Handler = {
+    if (ctx.ON != null) {
```

Comment:
it is not necessary to check if ctx.ON is not null, since it is a token that must occur in this grammar (similarly for RESUME)